xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ExportAction.java

```diff
@@ -197,23 +220,69 @@ private String exportHTML(XWikiContext context) throws XWikiException, IOExcepti
                     for (String docName : docsNames) {
                         String pageReference = wikiName + XWikiDocument.DB_SPACE_SEP + docName;
                         if (context.getWiki().getRightService().hasAccessLevel(
-                            "view", context.getUser(), pageReference, context))
+                                "view", context.getUser(), pageReference, context))
                         {
                             pageList.add(resolver.resolve(pageReference));
                         }
                     }
                 }
             } catch (QueryException e) {
                 throw new XWikiException(XWikiException.MODULE_XWIKI_APP, XWikiException.ERROR_XWIKI_APP_EXPORT,
-                    "Failed to resolve pages to export", e);
+                        "Failed to resolve pages to export", e);
             } finally {
                 context.setWikiId(database);
             }
         }
-
         return pageList;
     }
 
+    /**
+     * Resolve the set of pages to export based on choices made in a paginated tree
+     * @param otherPages if true, the pagination was not completely resolved and the "other pages" checkbox
+     *                  was checked: the hidden pages should be taken into account
+     * @param checkedPages gives the name of the pages purposely selected (only used if otherPages is false)
+     * @param uncheckedPages gives the name of the pages purposely deselected (only used if otherPages is true)
+     * @param context the context of the request
+     * @return the collection of document reference to export
+     */
+    private Collection<DocumentReference> resolvePagesToExport(boolean otherPages, String[] checkedPages, String[] uncheckedPages, XWikiContext context)
+        throws XWikiException
+    {
+        if (otherPages) {
+            // if other pages is selected, we select all but the unchecked pages
+            XWikiDocument doc = context.getDoc();
+
+            // use the document tree to get all the pages
+            Tree nestedpages = Utils.getComponent(Tree.class, "nestedPages");
+            String nodeId = "document:" + doc.getDocumentReference().toString();
+            int childCount = nestedpages.getChildCount(nodeId);
+            List<String> children = nestedpages.getChildren(nodeId, 0, childCount);
+
+            // result as obtained as nodeId with "document:" as prefix
+            List<String> formattedReferences = new ArrayList<>();
+            for (String child : children) {
+                int index = "document:".length();
+                formattedReferences.add(child.substring(index));
+            }
+
+            // we need to add the current document also
```

Comment:
Why?

Reply:
I think the reason is that currently you can't select "other pages" without selecting the parent page.